docs/user-guide/deployments.md

```diff
@@ -454,6 +454,133 @@ nginx-deployment-3066724191   0         0         1h
 Note: You cannot rollback a paused Deployment until you resume it.
 
 
+## Failed Deployment
+
+There will be times your Deployment will get stuck trying to deploy its newest ReplicaSet without ever completing,
+due to insufficient quota, readiness probe failures, image pull error, insufficient permissions, limit ranges, or any
+application runtime misconfiguration. In those cases you can specify a timeout parameter in the spec
+([spec.progressDeadlineSeconds](#progress-deadline-seconds)) that denotes the seconds you want to wait for your Deployment
+to report any progress. Once the deadline is exceeded, meaning your Deployment will not report any progress for more than
+progressDeadlineSeconds, the deployment controller will add a condition with `Type=Progressing`, `Status=False`, and
+`Reason=ProgressDeadlineExceeded` in the status of the Deployment. For more information on status conditions, see
+[here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/devel/api-conventions.md#typical-status-properties).
+
+Note that at the moment, Kubernetes will do nothing more than just report with a condition with `Reason=ProgressDeadlineExceeded`.
+In the future, once we support automatic rollbacks and users have opted-in, the deployment controller will rollback as
+soon as a failure condition is noticed.
+
+Another thing to note is transient errors eg. your rollout times out and a condition with `Reason=ProgressDeadlineExceeded`
+is set to your Deployment because of insufficient quota. All you need is to scale down your Deployment, or scale down other
+controllers you may be running, or increase quota in your namespace, either by asking your administrator or by doing it yourself
+assuming you have the right permissions. Once the deployment controller notices that your Deployment is complete, it will
+transition the status of the Progressing condition into a succeeded one (`Status=True` and `Reason=NewReplicaSetAvailable`).
```

Comment:
The sentence "All you need..." is very difficult to read.  Suggest breaking it into two: "To fix this issue, scale down your Deployment, scale down other controllers you may be running, or increase quota in your namespace.  You can do this by either by asking your administrator or doing it yourself assuming you have the right permissions."